actions/java-gradle-test/action.yaml

```diff
@@ -0,0 +1,88 @@
+
+inputs:
+  integration-tests:
+    description: "If true, the action is executing integration tests (Default is true)"
+    required: false
+    default: true
+  sonar-token:
+    description: "Token for sonarcloud."
+    required: false
+  signing-secret-key-ring:
+    description: "Key ring file for signing the sonatype publication."
+    required: false
+  signing-key-id:
+    description: "Key id for signing the sonatype publication."
+    required: false
+  signing-password:
+    description: "Password for signing the sonatype publication."
+    required: false
+  java-distribution:
+    description: "Java distribution to be installed. (Default is temurin)"
+    required: false
+    default: "temurin"
+  java-version:
+    description: "Java version to be installed. (Default is 11)"
+    required: false
+    default: "11"
+  gradle-version:
+    description: "Gradle version to be installed. (Default is 6.7.1)"
+    required: false
+    default: "6.7.1"
+  gradle-cache:
+    description: "Cache to use. Options are gradle, maven, sbt. (Default is '')"
+    required: false
+    default: ""
+  working-directory:
+    description: "Working directory of your Gradle artifacts. (Default is root directory)"
+    required: false
+    default: "./"
+runs:
+  using: "composite"
+  steps:    
+    - name: Check out repository
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    
+    - name: Set up Gradle with version ${{ inputs.gradle-version }}
+      uses: bakdata/ci-templates/actions/java-gradle-setup@feature/java-gradle
+      with:
+        java-distribution: ${{ inputs.java-distribution }}
+        java-version: ${{ inputs.java-version }}
+        gradle-version: ${{ inputs.gradle-version }}
+        gradle-cache: ${{ inputs.gradle-cache }}
+
+    - name: Run unit tests
+      run: ./gradlew check --info --stacktrace -PexcludeTests="**/*IntegrationTest*"
+      shell: bash
+      working-directory: ${{ inputs.working-directory }}
+    
+    - name: Run integration tests
+      if: ${{ inputs.integration-tests == 'true' }}
+      run: ./gradlew check --info --stacktrace -PincludeTests="**/*IntegrationTest*"
+      shell: bash
+      working-directory: ${{ inputs.working-directory }}
+    
+    # sonar.branch.autoconfig.disabled is necessary for autocreation of projects on sonarcloud to work
+    - name: Assess code quality
+      if: ${{ inputs.sonar-token != '' }}
+      run: ./gradlew -Dsonar.branch.autoconfig.disabled=true -Dsonar.host.url=https://sonarcloud.io -Dsonar.organization=bakdata --info --stacktrace sonarqube
+      shell: bash
+      working-directory: ${{ inputs.working-directory }}
+      env:
+        SONAR_TOKEN: ${{ inputs.sonar-token }}
+    
+    - name: Check if everything builds (incl. javadoc) for snapshot/release and signing works
+      if: ${{ inputs.signing-secret-key-ring != '' && inputs.signing-key-id != '' && inputs.signing-password != '' }}
```

Comment:
I don't really like this if-statement logic. I think this will get really confusing for the end-users. Besides, your action description only points out that this action is testing your...

Reply:
Same as above. This does not really have a better place and it tests the signing, so it kind of belongs to testing, but it is optional because not everybody wants to use Sonatype. I am open for suggestions

Reply:
Resolving because there does not seem to be a better place for it